server/src/test/java/io/envoyproxy/controlplane/server/V2DiscoveryServerAdsWarmingClusterIT.java

```diff
@@ -138,23 +135,32 @@ public void validateTestRequestToEchoServerViaEnvoy() throws InterruptedExceptio
             .and().body(containsString(UPSTREAM.response)));
   }
 
-  private static void createSnapshotWithWorkingClusterWithTheSameEdsVersion(DiscoveryRequest request,
-                                                                            ExecutorService executorService) {
+  private static void createSnapshotWithWorkingClusterWithTheSameEdsVersion(
+      DiscoveryRequest request,
+      ExecutorService executorService
+  ) {
     if (request.getTypeUrl().equals(Resources.V2.CLUSTER_TYPE_URL)) {
-      executorService.submit(() -> cache.setSnapshot(
-          GROUP,
-          createSnapshot(true,
-              "upstream",
-              UPSTREAM.ipAddress(),
-              EchoContainer.PORT,
-              "listener0",
-              LISTENER_PORT,
-              "route0",
-              "2"))
+      executorService.submit(() -> {
+            try {
+              Thread.sleep(5000);
```

Comment:
i may be missing a point, but why do we need to delay this snapshot by 5 sec?

Reply:
Just to make sure that Envoy "hits" this: https://github.com/envoyproxy/java-control-plane/pull/145/files#diff-255fe1430261e6c0881f777cc6b39d8cR89 (a badly configured Endpoint) . The tests sometimes "flew through" without Envoy returning any 503. What we actually could do is make an assertion that we receive one 503, send the snapshot and then assert 200. WDYT?

Reply:
i think using `503` is a better way to express the intention+ using `Thread.sleep` in tests cold be tricky on some platforms